lib/popularity.ts

```diff
@@ -0,0 +1,9 @@
+import {SitClass} from "../types/sitTypes";
+import {ClassPopularity} from "../types/derivedTypes";
+
+export function determineClassPopularity(sitClass: SitClass) {
+    if (!sitClass || sitClass.available === undefined) return ClassPopularity.Unknown
+    if (sitClass.available <= 0) return ClassPopularity.High
+    if (sitClass.available / sitClass.capacity >= 0.8) return ClassPopularity.Medium
```

Comment:
I think this would be the proper implementation of the Medium rule you described
```suggestion
    if (sitClass.available / sitClass.capacity <= 0.2) return ClassPopularity.Medium
```